src/Models.js

```diff
@@ -5,11 +5,12 @@
   LifeGamification.skillsCollection = skillsCollection;
 
   class Skill{
-    constructor(name, exp){
+    constructor(name, exp, timerData){
       this.name = name;
       this.exp = exp;
       this.calcLevel();
       this.calcExpTillNextLevel();
+      this.timer = new Timer(timerData);
```

Comment:
```javascript
if (timerData === null || timerData === {}) {
  timerData = {
     history: {},
     startTime: null
  }
}
this.timer = new Timer(timerData);
```

Reply:
```javascript
if (!timerData) {...}
```
Handles: timerData being ```null```, ```0```, ```{}``` or ```undefined```. If ```skillData``` doesn't contain ```timer```, ```timerData``` in ```Skill``` constructor will be ```undefined``` and our solution requires handling this case, so I will implement it like that.